File: Dataset/Leetcode/train/12/208.java
 class Solution {
    public String XXX(int num) {
        String ans;
        String[] roma1 = new String[]{"I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
        String[] roma2 = new String[]{"X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
        String[] roma3 = new String[]{"C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
        // 个位数
        int i = num % 10;
        if (i == 0) ans = "";
        else ans = roma1[i - 1];
        // 十位数
        int j = num / 10 % 10;
        if (j != 0)
            ans = roma2[j - 1] + ans;
        // 百位数
        int k = num / 100 % 10;
        if (k != 0)
            ans = roma3[k - 1] + ans;
        // 千位数
        int l = num / 1000 % 10;
        if (l != 0)
            for (int m = 0; m < l; m++) {
                ans = "M" + ans;
            }
        return ans;
    }
}

File: Dataset/Leetcode/train/13/208.cpp
class Solution {
public:
    int XXX(string s) {
    if(s.find("IV")!=s.npos)
		s = s.replace(s.find("IV"),2,"a");
	if (s.find("IX") != s.npos)
		s = s.replace(s.find("IX"), 2, "b");
	if (s.find("XL") != s.npos)
		s = s.replace(s.find("XL"), 2, "c");
	if (s.find("XC") != s.npos)
		s = s.replace(s.find("XC"), 2, "d");
	if (s.find("CD") != s.npos)
		s = s.replace(s.find("CD"), 2, "e");
	if (s.find("CM") != s.npos)
		s = s.replace(s.find("CM"), 2, "f");
	int ans = 0;
	for (int i = 0; i < s.size(); i++) {
		switch (s[i]) {
		case 'I':
			ans += 1;
			break;
		case 'V':
			ans += 5;
			break;
		case 'X':
			ans += 10;
			break;
		case 'L':
			ans += 50;
			break;
		case 'C':
			ans += 100;
			break;
		case 'D':
			ans += 500;
			break;
		case 'M':
			ans += 1000;
			break;
		case 'a':
			ans += 4;
			break;
		case 'b':
			ans += 9;
			break;
		case 'c':
			ans += 40;
			break;
		case 'd':
			ans += 90;
			break;
		case 'e':
			ans += 400;
			break;
		case 'f':
			ans += 900;
			break;
		}
	}
	return ans;
    }  
};

